pkg/xds/context/mesh_context_builder.go

```diff
@@ -553,6 +553,9 @@ func inferServiceProtocol(endpoints []xds.Endpoint) core_mesh.Protocol {
 	}
 	for _, endpoint := range endpoints[1:] {
 		endpointProtocol := core_mesh.ParseProtocol(endpoint.Tags[mesh_proto.ProtocolTag])
+		if endpoints[0].ExternalService != nil && endpoints[0].ExternalService.Protocol != "" {
+			endpointProtocol = endpoints[0].ExternalService.Protocol
+		}
```

Comment:
It doesn't make sense to loop if we're always checking index [0].

```suggestion
```

shouldn't we just return the following in line 552

```
return endpoints[0].ExternalService.Protocol
```

?

why does it make sense to first assign `serviceProtocol = endpoints[0].ExternalService.Protocol` and then assign `endpointProtocol = endpoints[0].ExternalService.Protocol` (which is the same) and do `GetCommonProtocol` on it?

Reply:
My bad it supposed to be `endpoint.ExternalService != nil && endpoint.ExternalService.Protocol`